File: clientsideYad2/src/models/product.ts
//import Owner from "./owner";

class Product {
  id: number;
  name: string = '';
  publishDate: Date;
  image: string = '';
  price: number;
  ownerId: number;
 // owner: Owner;

  constructor(
    id: number = 0,
    name: string = '',
    publishDate: Date = new Date(),
    image: string = '',
    price: number = 0,
    ownerId: number = 0,
    //owner: Owner = new Owner()
  ) {
    this.id = id;
    this.name = name;
    this.publishDate = publishDate;
    this.image = image;
    this.price = price;
    this.ownerId = ownerId;
   // this.owner = owner;
  }
}

export default Product;

File: Server_YAD2/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Server_YAD2.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime PublishDate { get; set; }
        public string Image { get; set; } = string.Empty;
        public double Price { get; set; }
        [ForeignKey(nameof(Owner.Id))]
        public int OwnerId { get; set; }
        

    }
}
